src/utils/auditorium.ts

```diff
@@ -0,0 +1,92 @@
+import {
+  SECTION_DEFAULT_COLUMNS_COUNT,
+  SECTION_DEFAULT_ROWS_COUNT,
+  SECTION_VIDEO_MIN_WIDTH_IN_SEATS,
+} from "settings";
+
+import { AuditoriumSection } from "types/auditorium";
+import { AuditoriumVenue } from "types/venues";
+import { AuditoriumSize } from "types/auditorium";
+import { User } from "types/User";
+
+import { WithId } from "./id";
+
+const emptyFilteredUsers: WithId<User>[] = [];
+
+export interface ConvertCoordinateProps {
+  index: number;
+  totalAmount: number;
+}
+
+export const convertToCartesianCoordinate = ({
+  index,
+  totalAmount,
+}: ConvertCoordinateProps) => index - Math.floor(totalAmount / 2);
+
+export const chooseAuditoriumSize = (sectionsCount: number) => {
+  if (sectionsCount <= 4) return AuditoriumSize.EXTRASMALL;
+
+  if (sectionsCount > 4 && sectionsCount <= 6) return AuditoriumSize.SMALL;
+
+  if (sectionsCount > 6 && sectionsCount <= 8) return AuditoriumSize.MEDIUM;
+
+  if (sectionsCount > 8 && sectionsCount <= 14) return AuditoriumSize.LARGE;
+
+  return AuditoriumSize.EXTRALARGE;
+};
+
+export const getVideoSizeInSeats = (columnCount: number) => {
+  // Video takes 1/3 of the seats
+  const videoWidthInSeats = Math.max(
+    Math.floor(columnCount / 3),
+    SECTION_VIDEO_MIN_WIDTH_IN_SEATS
+  );
+
+  // Keep the 16:9 ratio
+  const videoHeightInSeats = Math.ceil(videoWidthInSeats * (9 / 16));
```

Comment:
You are right, it's -1 row according to my calculations. Probably because of this logic:

![image](https://user-images.githubusercontent.com/63194656/124928909-b2db2a80-e008-11eb-82d6-92f32abe6696.png)